### ISSUE TITLE
Oracle does not receive the queries 

### ISSUE THREAD

**Issue by david179** [oraclize/ethereum-bridge#68]
I have a private network with a few nodes and I have installed ethereum-bridge to deploy an Oracle. 

One of my blockchain nodes is used by ethereum-bridge to deploy the oracle. The node is started with the command
`geth.exe --identity "7"  --datadir "D:\Blockchain\1" --networkid 4999 --rpc --rpcport 47000 --rpcapi "db,eth,net,web3,shh,personal" --rpccorsdomain "*"  --port 30800 --nodiscover  --shh --ipcdisable console`

while `ethereum-bridge`:

`ethereum-bridge -H localhost:47000 -a 0`

These are the logs on the `ethereum-bridge` console

```
[2019-02-21T09:45:14.820Z] INFO you are running ethereum-bridge - version: 0.6.1
[2019-02-21T09:45:14.821Z] INFO saving logs to: ./bridge.log
[2019-02-21T09:45:14.822Z] INFO using active mode
[2019-02-21T09:45:14.823Z] INFO Connecting to eth node http://localhost:47000
[2019-02-21T09:45:16.101Z] INFO connected to node type Geth/7/v1.8.21-stable-9dc5d1a9/windows-amd64/go1.11.4
[2019-02-21T09:45:16.588Z] WARN Using 0x8668dc605fe8d721dbb96378fd8155c621bea37d to query contracts on your blockchain, make sure it is unlocked and do not use the same address to deploy your contracts
[2019-02-21T09:45:16.711Z] INFO deploying the oraclize connector contract...
[2019-02-21T09:45:26.998Z] INFO connector deployed to: 0x251deddf98aca280a252d9e5f7b1b246bc3f66f7
[2019-02-21T09:45:27.119Z] WARN deterministic OAR disabled/not available, please update your contract with the new custom address generated
[2019-02-21T09:45:27.120Z] INFO deploying the address resolver contract...
[2019-02-21T09:45:37.534Z] INFO address resolver (OAR) deployed to: 0xb69156280a1461003a079387c2b5b827b3f0fc53
[2019-02-21T09:45:37.535Z] INFO updating connector pricing...
[2019-02-21T09:45:47.861Z] INFO successfully deployed all contracts
[2019-02-21T09:45:47.871Z] INFO instance configuration file saved to D:\GitHub\ethereum-bridge\config\instance\oracle_instance_20190221T094547.json

Please add this line to your contract constructor:

OAR = OraclizeAddrResolverI(0xB69156280a1461003a079387c2B5B827B3f0fc53);

[2019-02-21T09:45:47.996Z] INFO Listening @ 0x251deddf98aca280a252d9e5f7b1b246bc3f66f7 (Oraclize Connector)

(Ctrl+C to exit)
```

Then, using a different blockchain node I have deployed the following contract:

```
pragma solidity ^0.5.4;

import "github.com/oraclize/ethereum-api/oraclizeAPI.sol";

contract WolframAlpha is usingOraclize {

    uint256 public beginningPastQuarter;
    uint256 public beginningCurrentQuarter;
    uint256 public payoutIntervalSeconds;
    
    event newOraclizeQuery(string description);
    event newTimestampMeasure(uint256 timestamp);

     constructor() public {
        payoutIntervalSeconds = 120;
        beginningPastQuarter = 0;
        beginningCurrentQuarter = 0;
        OAR = OraclizeAddrResolverI(0xB69156280a1461003a079387c2B5B827B3f0fc53);
    }
    
    function stringToUint(string memory s) internal pure returns (uint) {
        bytes memory b = bytes(s);
        uint result = 0;
        for (uint i = 0; i < b.length; i++) { 
            if (uint8(b[i]) >= 48 && uint8(b[i]) <= 57) {
                result = result * 10 + (uint8(b[i]) - 48); 
            }
        }
        return result; 
    }

    function __callback(bytes32 myid, string memory result) public {
        if (msg.sender != oraclize_cbAddress()) revert();
        
        uint256 t = stringToUint(result);
        if (t-beginningCurrentQuarter >= payoutIntervalSeconds){
            beginningPastQuarter = beginningCurrentQuarter;
            beginningCurrentQuarter = t;
        }
        
        emit newTimestampMeasure(t);
    }
    
    function getUTCTimestamp(uint256 delay) public payable {
        emit newOraclizeQuery("Oraclize query was sent, standing by for the answer..");
        oraclize_query(delay,"WolframAlpha", "Timestamp now");
    }
    
    function pay(address receiver) public payable{
        address(uint160(receiver)).send(msg.value);
    }
    
    function balance(address a) public returns(uint256 r){
        return a.balance;
    }
} 
```

After deploying the contract, I have sent some ether to it so it can pay for the calls to the Oracle. 

Now if I try to use the oracle by calling `contractInstance.getUTCTimestamp(0,{from:eth.accounts[0],value:2222222,gas:5000000});` the transaction is mined but nothing happens in the `ethereum-bridge` console. 

If I inspect the transaction and its receipts I get:

```
{
  blockHash: "0x1c2d77b1e07d51894fdf0e2c3423d01538149d657b89e2568b5846d049452fda",
  blockNumber: 5799,
  contractAddress: null,
  cumulativeGasUsed: 5000000,
  from: "0xed2632f90f57dbbeb8e1d27277f4ef85a42851d2",
  gasUsed: 5000000,
  logs: [],
  logsBloom: "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  root: "0x67de497db6efcd76b0f4aba9255cd7744430eb7aded05e86090952071c53e97a",
  to: "0x217e753369532055c55c3588774275b8a58d5cf4",
  transactionHash: "0x21fb0191b15e748f93e9884349f8066cd765c0948d8050d99ebe99d32eb9543b",
  transactionIndex: 0
}
```

```
{
  blockHash: "0x1c2d77b1e07d51894fdf0e2c3423d01538149d657b89e2568b5846d049452fda",
  blockNumber: 5799,
  from: "0xed2632f90f57dbbeb8e1d27277f4ef85a42851d2",
  gas: 5000000,
  gasPrice: 20000000000,
  hash: "0x21fb0191b15e748f93e9884349f8066cd765c0948d8050d99ebe99d32eb9543b",
  input: "0xa01569d70000000000000000000000000000000000000000000000000000000000000000",
  nonce: 1048,
  r: "0x782a4d75ca59f5f0646733653474c6a42e26001513996c86635be1394d23d1c2",
  s: "0x23a1346876b104f527e291f885bfb21fe406b17128a18ddb7167bdf8ce021555",
  to: "0x217e753369532055c55c3588774275b8a58d5cf4",
  transactionIndex: 0,
  v: "0x41",
  value: 2222222
}
```

It seems to me that the transaction is using all gas and that happens even if I increase the gas value. The weird thing is that if I test this contract in remix the total gas used is below 200000. What could be the problem?

**Comment by gskapka** [comment]
What _address_ are you deploying your Oraclize-using contracts with?

**Comment by david179** [comment]
The Oracle contracts use the address `0x8668dc605fe8d721dbb96378fd8155c621bea37d` while the Oraclize-using one uses `0xed2632f90f57dbbeb8e1d27277f4ef85a42851d2`

**Comment by gskapka** [comment]
Is the first account in your `keys.json` file unlocked? 

Are the nodes in your private network local to the same machine? And if so, are their ports (both eth & rpc) unique?

**Comment by david179** [comment]
I have been using the active mode and so ethereum-bridge uses the accounts that exist already and so I don't have a  `keys.json` file in `ethereum-bridge\config\instance`. Yes the accounts are unlocked. I have tried to used the broadcast option which would use the `keys.json` file but it doesn't work.

**Comment by david179** [comment]
I have been using the active mode and so ethereum-bridge uses the accounts that exist already and so I don't have a  `keys.json` file in `ethereum-bridge\config\instance`. Yes the accounts are unlocked. 
I have tried to used the broadcast mode but that doesn't work for me. I get these errors:

```
ethereum-bridge -H localhost:27000 --broadcast --new
Please wait...
[2019-02-21T16:27:42.747Z] WARN keys.json not found, creating the new file D:\GitHub\ethereum-bridge\config\instance\keys.json
[2019-02-21T16:27:42.749Z] INFO you are running ethereum-bridge - version: 0.6.1
[2019-02-21T16:27:42.749Z] INFO saving logs to: ./bridge.log
Error: Error: Error: Keystore: Must include hdPathString in createVault inputs. Suggested alternatives are m/0'/0'/0' for previous lightwallet default, or m/44'/60'/0'/0 for BIP44 (used by Jaxx & MetaMask)
    at D:\GitHub\ethereum-bridge\bridge.js:201:22
    at D:\GitHub\ethereum-bridge\lib\bridge-util.js:39:21
    at Function.KeyStore.createVault (D:\GitHub\ethereum-bridge\node_modules\eth-lightwallet\lib\keystore.js:98:12)
    at Object.generateNewAddress (D:\GitHub\ethereum-bridge\lib\bridge-util.js:36:22)
    at Object.<anonymous> (D:\GitHub\ethereum-bridge\bridge.js:200:16)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
    at startup (internal/bootstrap/node.js:282:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)

Please wait...
D:\GitHub\ethereum-bridge\lib\blockchain-interface.js:32
  this.mainNode = config.node.main || '127.0.0.1:8545'
                         ^

TypeError: Cannot read property 'node' of undefined
    at BlockchainInterface [as constructor] (D:\GitHub\ethereum-bridge\lib\blockchain-interface.js:32:26)
    at singletonWrapper (D:\GitHub\ethereum-bridge\node_modules\pragma-singleton\PragmaSingleton.js:61:16)
    at process.removeAllLogs (D:\GitHub\ethereum-bridge\lib\bridge-log-manager.js:146:7)
    at process.emit (events.js:182:13)
    at process._fatalException (internal/bootstrap/node.js:493:27)

```

**Comment by D-Nice** [comment]
Since you mentioned Remix, when deploying on Remix, is optimization turned on? Which would explain the gas difference, in which case, you may wish to deploy with more gas, as the simplest solution, or ensure the deployed contract is optimized.

In general, have you tried just increasing the gas of your deployment transaction to the maximum block gas limit?

**Comment by david179** [comment]
Yes I used optimization in Remix, but I think that the gas use should be the same also in my private network because I downloaded the bytecode and ABI from Remix. 
To deploy the contract I used these calls and the gas is almost equal to the limit
```
var deploy = {from:eth.accounts[1],data:bytecode,gas:30000000};
var partial_instance = contract.new(deploy);

// after the block is mined
var contractInstance = contract.at(partial_instance.address);
```
I have tried to increase the gas also in the call

**Comment by D-Nice** [comment]
> Is it normal that gasPrice is greater than gas in ` eth.getTransaction("0x...");`?

Yes, gasPrice should normally be within gigawei units, the default is normally 20 gwei and that is how the gas is priced. The block gas limit is 4-8M normally depending on setup (on your privatechain you have full control over the bounds of both technically).

Note, that if you used the JavascriptVM on Remix, it's not 1:1 with a real network.

**Comment by david179** [comment]
I have created a new blockchain with an higher gas limit. I increased it from `0x2100000` to `0x4100000000` but the problem persists. No matter how much gas I put in the calls to the Oracle using smart contract it is always used completely.

**Comment by D-Nice** [comment]
Duplicate privatenet issue, refer to https://github.com/oraclize/ethereum-bridge/issues/73#issuecomment-488338183

**Comment by IamChuancey** [comment]
The problem is solved in  #73
